code/ensemble/methods/create-ensemble-relative-skill.R

```diff
@@ -0,0 +1,77 @@
+# Ensemble by relative skill
+#
+# Ensemble = sum of forecast values weighted by the inverse of relative skill
+# Weights are by model, horizon, target, location
+# i.e. not weighted by quantile
+#'
+#' @param forecasts forecasting models used for ensemble
+#' @param forecast_date date with saved evaluation csv of forecasts
+#' @param continuous_weeks include only forecasts with a history of evaluation
+
+library(dplyr)
+
+create_ensemble_relative_skill <- function(forecasts,
+                                           evaluation_date,
+                                           continuous_weeks = 4,
+                                           return_criteria = FALSE,
+                                           verbose = FALSE) {
+
+# Get evaluation ----------------------------------------------------------
+  evaluation <- try(suppressMessages(
+    vroom(here("evaluation", paste0("evaluation-", evaluation_date, ".csv")))))
+  # evaluation error catching
+  if ("try-error" %in% class(evaluation)) {
+    stop(paste0("Evaluation not found for ", evaluation_date))
+  }
+  if (verbose) {message(paste0("Relative skill evaluation as of ",
+                               evaluation_date))}
+  if (!"relative_skill" %in% names(evaluation)) {
+    stop("Evaluation does not include relative skill")
+  }
+
+  # include only models with forecasts,
+  #   with evaluation for >= x weeks
+  skill <- evaluation %>%
+    select(model, continuous_weeks, target_variable,
+           horizon, location, relative_skill) %>%
+    filter(model %in% forecasts$model &
+             continuous_weeks >= !!continuous_weeks &
+             !is.na(relative_skill))
+
+# Find weights ---------------------------------------------
+  # Take inverse of relative skill
+  skill <- skill %>%
+    mutate(inv_skill = ifelse(relative_skill > 0,
+                              1/relative_skill, 0))
+
+  # Weights for each model, horizon, location, target
+  skill <- skill %>%
+    group_by(target_variable, location, horizon) %>%
```

Comment:
can we make grouping by horizon optional? I don't think it's a bad idea as such, but having varying weights by horizon means that one might get pathological behaviour (e.g., uncertainty decreasing with increasing horizon). If not grouping by horizon one could average the skill over all horizons first by model, location and target variable and then apply the weighting by location and target variable.